Added a Request Body textarea along with a corresponding Content-Type…

philip opened the pull request:
Added a Request Body textarea along with a corresponding Content-Type definition.
Code taken directly from the following fork:
  https://github.com/taggiasco/resttesttest
Patches in that fork included many other changes, so can't provide a direct link.

philip committed:
Added a Request Body textarea along with a corresponding Content-Type definition.
Code taken directly from the following fork:
  https://github.com/taggiasco/resttesttest
Patches in that fork included many other changes, so can't provide
a direct link.

philip commented:
This pull request also helps with Issue #7

taggiasco commented:
Hi,
How do you deal with parameters ? Since you have a body, parameters are ignored for the moment.
In my fork, I added a "buildArguments" which appends the parameters to the URL (only when you have a body). Not sure if it is the right way of doing this, but the request seems to be correctly handled by the test service (https://httpbin.org/put).
Best regards,
Matt

philip commented:
@taggiasco Ah, an excellent point, thank you. My attempts to keep the pull request as simple/small as possible missed that fact, and my tests all used headers+body and not parameters. I'll update the pull request at some point.

philip committed:
Missed the buildArguments part in @taggiasco 's patch. Now, when a
request body is sent, the parameters are also included.

Still seems to cause issues when using GET + Body, will look into
if that should be allowed and/or handle the error more gracefully.

philip commented:
@taggiasco @jeroenooms Okay, this merge request seems fine now. I don't fully understand this topic yet as I'm still learning the ins and outs of REST APIs (and HTTP requests in general).

bracco23 commented:
Is there any plan to finally merge this and add this functionality?

philip commented:
@bracco23 This project is no longer active (no commits in 2+ years) so I doubt it.

jeroen commented:
You know you can easily deploy your own version right? It's just static html. You can upload it to anywhere you want and it'll work :)

philip commented:
@jeroen True, although it's easier to click a button (download) than worry about applying patches :) But I suspect you'll not merge this PR as to keep https://resttesttest.com/ as is, right? Understandable, of course.

redguyrichard committed:
Added presets, reset button, query string base64 encoding, spinner icon, response code and result in new tab links

philip committed:
Merge pull request #1 from redguyrichard/gh-pages

Added presets + more
